lib/ansible/executor/module_common.py

```diff
@@ -587,52 +587,69 @@ def _slurp(path):
     return data
 
 
-def _get_shebang(interpreter, task_vars, templar, args=tuple()):
+def _get_shebang(interpreter, task_vars, templar, args=tuple(), remote_is_local=False):
     """
     Note not stellar API:
        Returns None instead of always returning a shebang line.  Doing it this
        way allows the caller to decide to use the shebang it read from the
        file rather than trust that we reformatted what they already have
        correctly.
     """
-    interpreter_name = os.path.basename(interpreter).strip()
-
     # FUTURE: add logical equivalence for python3 in the case of py3-only modules
 
-    # check for first-class interpreter config
-    interpreter_config_key = "INTERPRETER_%s" % interpreter_name.upper()
+    interpreter_name = os.path.basename(interpreter).strip()
 
-    if C.config.get_configuration_definitions().get(interpreter_config_key):
-        # a config def exists for this interpreter type; consult config for the value
-        interpreter_out = C.config.get_config_value(interpreter_config_key, variables=task_vars)
-        discovered_interpreter_config = u'discovered_interpreter_%s' % interpreter_name
+    # name for interpreter var
+    interpreter_config = u'ansible_%s_interpreter' % interpreter_name
+    # key for config
+    interpreter_config_key = "INTERPRETER_%s" % interpreter_name.upper()
 
-        interpreter_out = templar.template(interpreter_out.strip())
+    interpreter_out = None
 
-        facts_from_task_vars = task_vars.get('ansible_facts', {})
+    # looking for python or variants
+    if 'python' in interpreter_name:
```

Comment:
This should still just probably be `==` to avoid any weird corner cases until/unless we fully support something else in the future.